File: Day 7/day7.py
def fix_packages(packages:str) -> str:
  while '(' in packages:
    idx_ini = packages.rfind('(')
    idx_fi = packages[idx_ini+1:].index(')') + 1 + idx_ini

    reversed_word = packages[idx_ini+1:idx_fi][::-1]
    word_parentesis = packages[idx_ini:idx_fi+1]
    packages = packages.replace(word_parentesis,reversed_word)

  return packages

File: Day 7/day7.ts
function fixPackages(packages:string): string {

    while (packages.includes('(')){
        const idxIni = packages.lastIndexOf('(');
        const idxFi  = packages.lastIndexOf(')', idxIni) + 1;

        const reversed_word = packages.slice(idxIni+1, idxFi).split('').reverse().join('');
        const word_parentesis = packages.slice(idxIni, idxFi);
        packages = packages.replace(word_parentesis,reversed_word);
    }

    return packages;
}